src/http-api/index.js

```diff
@@ -0,0 +1,37 @@
+var Hapi = require('hapi')
+var IPFS = require('../ipfs-core')
+var debug = require('debug')
+var log = debug('api')
+
+exports = module.exports
+
+exports.start = function (callback) {
+  // start IPFS and exports.ipfs = new IPFS()
+
+  exports.ipfs = new IPFS()
+
+  var server = exports.server = new Hapi.Server({
+    connections: {
+      routes: {
+        cors: true
+      }
+    }
+  })
+
+  server.connection({
+    port: 9000
+  })
+
+  // load routes
+  require('./routes/version.js')
```

Comment:
we could DI the server obj (and the IPFS core object), but I kind of like making available by knowing where to pick it from other then move it around. I've been doing this for a while, if it becomes complicated over time, I will reconsider your suggestion :)